manualconfig/routertopology/aggregator/readme.md

```diff
@@ -0,0 +1,250 @@
+# Aggregate Pattern using AMQ Interconnect routers & brokers
+
+## IPs used in this guide
+|Component|IP|
+|---|---|
+|Producer|192.168.56.30|
+|Consumer|192.168.56.31|
+|Aggregate|192.168.56.32|
+|Broker|192.168.56.33|
+
+## Configure Aggregate
+* Edit the qdrouterd.conf file
+```shell
+sudo vi /etc/qpid-dispatch/qdrouterd.conf
+```
+
+* Update the router section as below to reflect that we are establishing a network of routers
+```text
+router {
+    mode: interior 
+    id: Router.A
+}
+```
+
+* Define following listeners
+```text
+# Listener for Clients
+listener {
+    host: 0.0.0.0
+    port: 5772
+    authenticatePeer: no
+    saslMechanisms: ANONYMOUS
+}
+
+# Listener for Inter-Router connections
+listener {
+    host: 0.0.0.0
+    port: 5773
+    authenticatePeer: no
+    role: inter-router
+}
+
+# Listener for AMQ Console
+listener {
+    name: amq-console
+    role: normal
+    host: 0.0.0.0
+    port: 5673
+    http: yes
+}
+```
+
+* Define following connectors for Brokers
+```text
+connector {
+    name: broker-connector-1
+    host: 192.168.56.33
+    port: 5672
+    role: route-container
+    saslMechanisms: ANONYMOUS
+}
+
+connector {
+    name: broker-connector-2
+    host: 192.168.56.33
+    port: 5772
+    role: route-container
+    saslMechanisms: ANONOYMOUS
+}
+```
+
+* Define the Waypoint for the broker queue as follows
+```text
+# Waypoint for the Broker Queue
+address {
+    prefix: SampleQueue
+    waypoint: yes
+}
+```
+
+* Create autolinks for the broker queues for both broker 1 and broker 2
+```text
+# Create Autolink to send messages to the broker 1
+autoLink {
+    address: SampleQueue
+    connection: broker-connector-1
+    direction: out
+}
+
+autoLink {
+    address: SampleQueue
+    connection: broker-connector-1
+    direction: in
+}
+
+# Create Autolink to send messages to the broker 2
+autoLink {
+    address: SampleQueue
+    connection: broker-connector-2
+    direction: in
+}
+
+autoLink {
+    address: SampleQueue
+    connection: broker-connector-2
+    direction: out
+}
+```
+
+* As the Aggregate is going to listen for incoming connections from producer and consumer, open the needed ports
+```shell
+sudo firewall-cmd --permanent --add-port={5001,5772,5773,5673}/tcp
```

Comment:
@rhkp what's the importance of port 5001? I can't see it anywhere in the Aggregator config.